dockstore-client/src/main/java/io/dockstore/client/cli/YamlVerify.java

```diff
@@ -0,0 +1,170 @@
+
+        // Check Workflows
+        List<YamlWorkflow> workflows = dockstoreYaml12.getWorkflows();
+        if (workflows != null) {
+            for (YamlWorkflow workflow : workflows) {
+                List<String> filePaths = workflow.getTestParameterFiles();
+                filePaths.add(workflow.getPrimaryDescriptorPath());
+                missingFiles.addAll(filesExist(filePaths, basePath));
+            }
+        }
+        // Check Tools
+        List<YamlWorkflow> tools = dockstoreYaml12.getTools();
+        if (tools != null) {
+            for (YamlWorkflow tool : tools) {
+                List<String> filePaths = tool.getTestParameterFiles();
+                filePaths.add(tool.getPrimaryDescriptorPath());
+                missingFiles.addAll(filesExist(filePaths, basePath));
+            }
+        }
+
+        // Check service
+        Service12 service = dockstoreYaml12.getService();
+        if (service != null) {
+            missingFiles.addAll(filesExist(service.getFiles(), basePath));
+        }
+        if (!missingFiles.isEmpty()) {
+            String errorMsg = "";
+            for (String missingFile: missingFiles) {
+                errorMsg += missingFile + '\n';
+            }
+            throw new ValidateYamlException("Your file structure has the following errors:\n" + errorMsg);
+        }
+    }
+
+    public static void dockstoreValidate(String path) throws ValidateYamlException {
+        Path workflowPath = Paths.get(path);
+
+        // Verify that the path is a valid directory
+        if (!Files.isDirectory(workflowPath)) {
+            throw new ValidateYamlException(ERROR_MESSAGE + workflowPath.toString() + " does not exist");
+        }
+
+        // Verify that .dockstore.yml exists
+        Path dockstoreYmlPath = Paths.get(path, DOCKSTOREYML);
+        if (!Files.exists(dockstoreYmlPath)) {
+            throw new ValidateYamlException(ERROR_MESSAGE + dockstoreYmlPath.toString() + " does not exist");
+        }
+
+        // Verify that .dockstore.yml is non-empty
+        File dockstoreYml = new File(dockstoreYmlPath.toString());
+        if (dockstoreYml.length() == 0) {
+            throw new ValidateYamlException(ERROR_MESSAGE + dockstoreYmlPath.toString() + " is empty");
+        }
+
+        // Load dockstoreYml into string
+        String dockstoreYmlString = "";
+        try {
+            dockstoreYmlString = Files.readString(dockstoreYmlPath);
+        } catch (IOException ex) {
+            // Unlikely to ever catch
+            throw new ValidateYamlException("Error reading " + dockstoreYmlPath.toString() + " :\n" + ex.getMessage());
+        }
+
+        // Verify that the Yaml is valid, however does not verify it is valid for dockstore
+        Yaml yaml = new Yaml();
+        Map<String, Object> data = null;
+        try {
+            data = yaml.load(Files.readString(dockstoreYmlPath));
+            out(dockstoreYmlPath + " is a valid yaml file");
```

Comment:
this should be a string constant and referenced here and in your tests

Reply:
I have changed all the tests so they use string constants defined mostly in YamlVerify